src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -26,48 +27,54 @@ import Button from "components/atoms/Button";
 import "./UserProfileModal.scss";
 
 type PropTypes = {
-  show: boolean;
-  onHide: () => void;
   zIndex?: number;
-  userProfile?: WithId<User>;
 };
 
-const UserProfileModal: React.FunctionComponent<PropTypes> = ({
-  show,
-  onHide,
+export const UserProfileModal: React.FunctionComponent<PropTypes> = ({
   zIndex,
-  userProfile,
 }) => {
   const venue = useSelector(currentVenueSelector);
 
   const { user } = useUser();
 
-  const chosenUserId = userProfile?.id;
-
   const { selectRecipientChat } = useChatSidebarControls();
 
+  const {
+    selectedUserProfile,
+    hasSelectedProfile,
+    closeUserProfileModal,
+  } = useProfileModalControls();
+
+  const chosenUserId = selectedUserProfile?.id;
+
   const openChosenUserChat = useCallback(() => {
     if (!chosenUserId) return;
 
     selectRecipientChat(chosenUserId);
     // NOTE: Hide the modal, after the chat is opened;
-    onHide();
-  }, [selectRecipientChat, onHide, chosenUserId]);
+    closeUserProfileModal();
+  }, [selectRecipientChat, closeUserProfileModal, chosenUserId]);
 
-  if (!userProfile || !chosenUserId || !user) {
+  if (!selectedUserProfile || !chosenUserId || !user) {
     return <></>;
   }
 
   // REVISIT: remove the hack to cast to any below
   return (
-    <Modal show={show} onHide={onHide} style={zIndex && { zIndex }}>
+    <Modal
+      show={hasSelectedProfile}
+      onHide={closeUserProfileModal}
+      style={zIndex && { zIndex }}
```

Comment:
[NB] I think we don't need the `zIndex &&` check here, as if `zIndex` is `undefined` then it won't 'output' any styles, and if the styles object is empty, it won't output any styles. So I believe this can just be:

```suggestion
      style={{ zIndex }}
```